src/Adapters/BlobAdapter.ts

```diff
@@ -135,8 +140,10 @@ export default class BlobAdapter implements IBlobAdapter {
 
             try {
                 const configBlob = await getConfigBlob();
-                // Handle any schema migrations to transform legacy data into valid data
-                handleMigrations(configBlob.data);
+                if (showVisualRulesFeature) {
+                    // Handle any schema migrations to transform legacy data into valid data
+                    handleMigrations(configBlob.data);
```

Comment:
do we need to do null check on data before passing to handleMigration?

Reply:
`handleMigration` methods have null checks in them